### DISCUSSION TITLE
custom shapes

### DISCUSSION THREAD

**Discussion by completely-unstable** [hneemann/Digital#1323] [Q&A]
so when setting the shape of a component to 'User Defined', and also the circuit is generic, is there a way to set the svg from a code block? for example if i made a custom and gate, and i had two svg files, "and.svg", and "and3.svg", i want to do something like this:
```
inps := args.inputs;

if (inps = 2) {
  settings.customShape = "and.svg";
}
if (inps = 3) {
  settings.customShape = "and3.svg";
}
```
is this possible?

**Comment by hneemann** [comment]
No! Custom shapes don't work with generic circuits if the number of inputs or outputs depends on the parameterization.
This is actually not a problem, as you can use a bus whose width is generic parameterized.

**Reply by completely-unstable** [reply]
well that doesnt really work if the inputs are already multi-bit or if the inputs are control signals they wouldnt make sense with a bus. like the built in gates take variable # of bits and variable # of inputs, sometimes i want to make my component do that too and make shapes for each configuration. or if i have a component that has a custom shape and i want to add an option to invert some of the inputs/outputs it would be nice to be able to add a custom shape with the not bubbles for when those are set.
